content/applications/sales/sales/invoicing/milestone.rst

```diff
@@ -2,32 +2,244 @@
+
+    Those tasks are then attached to a pre-existing :guilabel:`Project`, which, in this case, is
+    titled, :guilabel:`Rebranding Projects`.
+
+To invoice milestones, create a sales order with the milestone product(s). To do that, go to
+:menuselection:`Sales app --> New`. Doing so reveals a blank quotation form.
+
+From this quotation form, add a :guilabel:`Customer`. Then, click :guilabel:`Add a product` in the
+:guilabel:`Order Lines` tab. Next, add the milestone product(s) to the :guilabel:`Order Lines` tab.
+
+Once the corresponding milestone product(s) have been added, click :guilabel:`Confirm` to confirm
+the order, which turns the quotation into a sales order.
+
+When the order is confirmed, new smart buttons appear at the top of the sales order based on what
+was selected in the :guilabel:`Create on Order` field on the product form.
+
+.. image:: milestone/confirmed-sales-order-milestone.png
+   :align: center
+   :alt: Confirmed sales order with milestone products in the order lines tab.
+
+From the sales order, click the :guilabel:`Milestones` smart button. Doing so reveals a populated
+:guilabel:`Milestones` page. Click :guilabel:`New` to add milestones.
+
+.. image:: milestone/adding-milestones.png
+   :align: center
+   :alt: Adding milestones to a sales order with milestone products.
+
+Enter a :guilabel:`Name` for the milestone. Next, apply it to the corresponding :guilabel:`Sales
+Order Item`. Lastly, assign a :guilabel:`Deadline` to the milestone, if desired.
+
+.. note::
+   The :guilabel:`Quantity` field is not editable.
+
+Repeat that process for all milestone sales order items.
+
+Then, return to the sales order, via the breadcrumbs. From the sales order, click the
+:guilabel:`Tasks` smart button. Doing so reveals a :guilabel:`Tasks` page with a task for each sales
+order item with that option designated in the :guilabel:`Create on Order` field.
+
+.. image:: milestone/tasks-page.png
+   :align: center
+   :alt: Sample tasks page accessed via the smart button from a sales order with milestone products.
+
+To manually assign a configured milestone to a task, click the desired task, which reveals the task
+form. On the task form, select the appropriate milestone to which this task should be connected, in
+the :guilabel:`Milestone` field.
+
+.. image:: milestone/milestone-field-on-task-form.png
+   :align: center
+   :alt: The milestone field on the task form when dealing with milestone products in Odoo Sales.
+
+Repeat this process for all milestone tasks.
+
+With those tasks properly configured, employees will log in their progress as they work on the task,
+in addition to adding any notes related to the task.
+
+Then, when that task is complete, that means that milestone has been reached. At that point, it is
+time to invoice that milestone.
+
+To do that, first return to the sales order -- either via the breadcrumb links, or by navigating to
```

Comment:
Totally optional formatting suggestion! It's just that Zac once replaced my double hypen (--) with an em dash (—), so I'll just do that here (:
```suggestion
To do that, first return to the sales order — either via the breadcrumb links, or by navigating to
```